services/inspector/inspector.go

```diff
@@ -33,14 +34,24 @@ type InspectorConfig struct {
 }
 
 func (ins *Inspector) Start() error {
-	for {
-		select {
-		case <-ins.ctx.Done():
-			return nil
-		case blockNum := <-ins.inspectCh:
-			ins.inspectionWorker(ins.ctx, blockNum)
-		}
+	if ins.cfg.Config.LocalModeConfig.Enable && !ins.cfg.Config.LocalModeConfig.EnableInspection {
+		log.Warn("inspection is disabled - please enable it from the local mode config if you need it")
+		return nil
 	}
+
+	ins.registerMessageHandlers()
+
+	go func() {
```

Comment:
I would rather go with the first one :slightly_smiling_face:  i don't feel comfortable having this interface in general (context as struct field, `Start` spawning a detached goroutine), but it is out of scope for this issue. 

pr looks good to me

Reply:
Thanks. I think we discussed the "detached goroutine" case before and you made some good points. The reason for making this async in here is because of the `StartServices` implementation here: https://github.com/forta-network/forta-node/blob/master/services/service.go#L145
- Services should attempt to start and return an error if they can't start for a reason so that we are able to handle the error on-the-fly.
- A service should start within acceptable time. Otherwise, that's a service start timeout error.
- All services should really start and in the order we declare. Otherwise, we fail the program.

`Start()` implementations and usage will need to be complicated if we intend to do `go Start()`.

Thinking of my washing machine or coffee machine for example: I just hit the start button and see if it has started okay, then leave. I expect to deal with the problems right that moment and don't expect that to call me back if it has a problem in starting. :slightly_smiling_face: